rust/sbp/src/de.rs

```diff
@@ -303,11 +306,14 @@ impl Frame {
 
     pub fn to_sbp(&self) -> Result<Sbp, Error> {
         self.check_crc()?;
-        Ok(Sbp::from_parts(
-            self.msg_type(),
-            self.sender_id(),
-            self.payload(),
-        )?)
+        let sender_id = self.sender_id();
+        let mut payload = self.payload();
+        let msg = Sbp::from_parts(self.msg_type(), sender_id, payload).or_else(|_| {
```

Comment:
I think this would mask malformed messages though? Like if the message type is a known message but the payload is too short. Then you'd have an `Unknown` message with a message type that corresponds to a known message which is a little odd. But maybe that's what we want, I guess it's a trade off between being lenient and surfacing errors better

Reply:
yea... think we're just treating Unknown as the `Frame` at this point to serialize as json :/

although that makes sense for the purpose of the Unknown message i guess.

maybe a better compromise would to just add a flag like `strict` to address dropping payload errors?

issue with the raw decoder or persisting broken data is where do we actually trim the unnecessary data - which thought everything would propagate up to sbp converters and they would drop those - or maybe we just never drop them 😱

Reply:
I think the `Decoder` should only output valid SBP (as in CRC is good and the payload is correct), if a user wants access to potentially invalid messages they can go with the `Framer` I think. So for example `sbp2json` should probably be doing something like

```rust
for frame in Framer::new(rdr) {
    if let Ok(msg) = frame.to_sbp() { 
        // valid crc/payload, possibly an Unknown message
        json_encoder.write(&msg)?;
    } else {
        // invalid crc or payload, could also be an Unknown message
        // need some json representation of a frame
        json_encoder.write(&frame)?;
    }
} 
```